File: data-structures/lists/python/nested_lists_delete.py
# Raspberry Pi Foundation
# Developed as part of Ada Computer Science
# 
# Usage licensed under CC BY-NC-SA 4.0


def delete_astronaut(all_astronauts, astronaut_index):
    """Delete an astronaut list at the given index"""

    del all_astronauts[astronaut_index]
    

def display_all_astronauts(all_astronauts):
    """Display the values of each astronaut in the nested list"""

    # Repeat for each astronaut in the outer list
    for astronaut in all_astronauts:
        # Repeat for each value in the inner list
        for value in astronaut:
            print(f"{value}  ", end = "") # Do not print new line
        print() # Print a new line after each astronaut


def main():
    # Declare and initialise the nested list of astronauts
    all_astronauts = [
        ["ISS", "Sergey Prokopyev", "176"],
        ["ISS", "Dmitry Petelin", "176"],
        ["ISS", "Frank Rubio", "176"],
        ["Tiangong", "Fei Junlong", "112"],
        ["Tiangong", "Deng Qingming", "107"],
        ["Tiangong", "Zhang Lu", "107"],
        ["ISS", "Stephen Bowen", "54"],
        ["ISS", "Warren Hoburg", "14"],
        ["ISS", "Sultan Al Neyadi", "14"],
        ["ISS", "Andrey Fedyaev", "14"]
    ]

    # Output the details of each astronaut in the list
    print("Details of each astronaut in the list:")
    display_all_astronauts(all_astronauts)

    # Delete the astronaut stored at index 0
    position = 0
    print(f"\nDelete the astronaut at position {position}")
    delete_astronaut(all_astronauts, position)

    # Output the details of each astronaut in the list
    print("\nDetails of each astronaut in the list:")
    display_all_astronauts(all_astronauts)


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == "__main__":
    main()

File: data-structures/lists/c-sharp/nested_lists_delete.cs
/*
Raspberry Pi Foundation
Developed as part of Ada Computer Science 
Usage licensed under CC BY-NC-SA 4.0

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;
using System.Collections.Generic;

namespace AdaCodeSamples {
    class NestedLists {
        // The Main method is the entry point for all C# programs
        public static void Main() {
            // Declare and initialise the nested list of astronauts
            List<List<string>> allAstronauts = new List<List<string>>() {
                new List<string>() { "ISS", "Sergey Prokopyev", "176" },
                new List<string>() { "ISS", "Dmitry Petelin", "176" },
                new List<string>() { "ISS", "Frank Rubio", "176" },
                new List<string>() { "Tiangong", "Fei Junlong", "112" },
                new List<string>() { "Tiangong", "Deng Qingming", "107" },
                new List<string>() { "Tiangong", "Zhang Lu", "107" },
                new List<string>() { "ISS", "Stephen Bowen", "54" },
                new List<string>() { "ISS", "Warren Hoburg", "14" },
                new List<string>() { "ISS", "Sultan Al Neyadi", "14" },
                new List<string>() { "ISS", "Andrey Fedyaev", "14" }
            };

            // Output the details of each astronaut in the list
            Console.WriteLine("\nDetails of each astronaut in the list:");
            DisplayAllAstronauts(allAstronauts);

            // Delete the astronaut stored at index 0
            int position = 0;
            Console.WriteLine($"\nDelete the astronaut at position {position}");
            DeleteAstronaut(allAstronauts, position);

            // Output the details of each astronaut in the list
            Console.WriteLine("\nDetails of each astronaut in the list:");
            DisplayAllAstronauts(allAstronauts);
        }

        // Delete an astronaut list at the given index
        public static void DeleteAstronaut(List<List<string>> allAstronauts,
            int astronautIndex) {

            allAstronauts.RemoveAt(astronautIndex);
        }

        // Display the values of each astronaut in the given list
        public static void DisplayAllAstronauts(List<List<string>> allAstronauts) {
            // Repeat for each astronaut in the outer list
            foreach (List<string> astronaut in allAstronauts) {
                // Repeat for each value in the inner list
                foreach (string value in astronaut) {
                    Console.Write($"{value}  "); // Do not print new line
                }
                Console.WriteLine(); // Print a new line after each astronaut
            }
        }
    }
}